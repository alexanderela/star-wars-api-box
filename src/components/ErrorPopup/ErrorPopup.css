.ErrorPopup {
  align-items: flex-start;
  display: flex;
  flex-direction: row;
  height: 2rem;
  justify-content: center; 
  margin: 0 0 3rem;
  overflow: hidden;
  z-index: 100;
}

.popup-inner {
  align-items: center;
  display: flex;
  justify-content: center;
}

h1 {
  color: #f9d500;
  font-size: 1.5rem;
  margin: 0;
}

button {

	cursor: pointer;
  color: #000;
  font-weight: 700;
  font-size: 0.75rem;
  background-color: #f9d500;
  border: none;
  border-radius: 10px;
  color: #000;
	cursor: pointer;
  font-weight: 700;
  margin: 0.1rem 0 0 0.3rem;
  padding: 0.35rem 0.5rem;
}


