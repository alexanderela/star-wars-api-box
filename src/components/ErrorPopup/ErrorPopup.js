import React, { Component } from 'react';
import './ErrorPopup.css';

class ErrorPopup extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="ErrorPopup">
        <div className="popup-inner">
          <h1>You have no favorites selected!</h1>
          <button
            onClick={this.props.closeError}
          >
X
          </button>
        </div>
      </div>
    );
  }
}

export default ErrorPopup;
