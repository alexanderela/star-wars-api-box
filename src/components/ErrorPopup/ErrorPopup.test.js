/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import ErrorPopup from './ErrorPopup';
import { shallow } from 'enzyme';

describe('ErrorPopup', () => {
	it('matches the snapshot', () => {
		const wrapper = shallow(<ErrorPopup />)
		expect(wrapper).toMatchSnapshot()
	})
})