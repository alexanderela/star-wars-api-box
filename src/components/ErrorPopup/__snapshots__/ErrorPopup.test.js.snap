// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPopup matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorPopup />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="popup-inner"
      >
        <h1>
          You have no favorites selected!
        </h1>
        <button>
          X
        </button>
      </div>,
      "className": "ErrorPopup",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            You have no favorites selected!
          </h1>,
          <button>
            X
          </button>,
        ],
        "className": "popup-inner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You have no favorites selected!",
          },
          "ref": null,
          "rendered": "You have no favorites selected!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "X",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="popup-inner"
        >
          <h1>
            You have no favorites selected!
          </h1>
          <button>
            X
          </button>
        </div>,
        "className": "ErrorPopup",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              You have no favorites selected!
            </h1>,
            <button>
              X
            </button>,
          ],
          "className": "popup-inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You have no favorites selected!",
            },
            "ref": null,
            "rendered": "You have no favorites selected!",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "X",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "X",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
